testplan/base.py

```diff
@@ -76,24 +78,170 @@ class Testplan(RunnableManager):
+    :param test_sorter: Tests sorting class.
+    :type test_sorter: Subclass of
+        :py:class:`BaseSorter <testplan.testing.ordering.BaseSorter>`
+    :param test_lister: Tests listing class.
+    :type test_lister: Subclass of
+        :py:class:`BaseLister <testplan.testing.listing.BaseLister>`
+    :param verbose: Enable or disable verbose mode.
+    :type verbose: ``bool``
+    :param debug: Enable or disable debug mode.
+    :type debug: ``bool``
+    :param timeout: Timeout value for test execution.
+    :type timeout: ``NoneType`` or ``int`` or ``float`` greater than 0.
+    :param interactive_handler: Handler for interactive mode execution.
+    :type interactive_handler: Subclass of :py:class:
+        `TestRunnerIHandler <testplan.runnable.interactive.TestRunnerIHandler>`  # pylint: disable=line-too-long
+    :param extra_deps: Extra module dependencies for interactive reload.
+    :type extra_deps: ``list`` of ``module``
     """
 
     CONFIG = TestplanConfig
 
-    def __init__(self, **options):
-        super(Testplan, self).__init__(**options)
+    def __init__(self,
+                 name,
+                 description=None,
+                 parse_cmdline=True,
+                 interactive=False,
+                 port=None,
+                 abort_signals=None,
+                 logger_level=logger.TEST_INFO,
+                 file_log_level=logger.DEBUG,
+                 runpath=path.default_runpath,
+                 path_cleanup=True,
+                 all_tasks_local=False,
+                 shuffle=None,
+                 shuffle_seed=float(random.randint(1, 9999)),
+                 exporters=None,
+                 stdout_style=defaults.STDOUT_STYLE,
+                 report_dir=defaults.REPORT_DIR,
+                 xml_dir=None,
+                 pdf_path=None,
+                 json_path=None,
+                 pdf_style=defaults.PDF_STYLE,
+                 report_tags=None,
+                 report_tags_all=None,
+                 merge_scheduled_parts=False,
+                 browse=None,
+                 ui_port=None,
+                 web_server_startup_timeout=defaults.WEB_SERVER_TIMEOUT,
+                 test_filter=filtering.Filter(),
+                 test_sorter=ordering.NoopSorter(),
+                 test_lister=None,
+                 verbose=False,
+                 debug=False,
+                 timeout=None,
+                 interactive_handler=TestRunnerIHandler,
+                 extra_deps=None,
+                 **options):
+
+        # Set mutable defaults.
+        if abort_signals is None:
+            abort_signals = []
+        if shuffle is None:
+            shuffle = []
+        if extra_deps is None:
+            extra_deps = []
+        if report_tags is None:
+            report_tags = []
+        if report_tags_all is None:
+            report_tags_all = []
+
+        super(Testplan, self).__init__(
+            name=name,
```

Comment:
can we use the filter_locals utils so that we don't copy them again?

Reply:
not really a big fan of using that - explicit is better than implicit ;)

Reply:
Actually, i'm having a second thought, we cannot really do the filter_locals directly here as it will eliminate pair with None-value. And user might explicitly set a cmdline option to None.

Reply:
OK yeah I'll leave it as is then.